### ISSUE TITLE
[java] SingularField ignores static variables

### ISSUE THREAD

**Issue by ghost** [pmd/pmd#3694]
**Affects PMD Version:**
6.41.0+
7.0.0 

**Rule:SingularField**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_design.html#singularfield>

**Description:**
Hi, I found two false negatives about the rule SingularField. Please refer to the following case, PMD should have reported a false negative at line 2 and 3. I think these cases are caused by modifiers of `var1` and `var2`.

**Code Sample demonstrating the issue:**
```java
class Impl {
  final private int var1 = 10;  // should report a warning here
  static private int var2 = 10; // also should report a warning here
  public int foo() {
    return var1;
  }
  public int foo(int y) {
    var2 = y + 5;
    return var2;
  }
}
```

**Expected outcome:**

PMD should report violations at line 2 and 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by adangel** [comment]
Actually, this is not fixed yet. The following test case still doesn't report any violations:

```xml
    <test-code>
        <description>[java] Make SingularField can detect modified variables #3694</description>
        <expected-problems>2</expected-problems>
        <expected-linenumbers>2,3</expected-linenumbers>
        <code><![CDATA[
class Impl {
  final private final int var1 = 10;  // should report a warning here
  static private int var2 = 10; // also should report a warning here
  public int foo() {
    return var1;
  }
  public int foo(int y) {
    var2 = y + 5;
    return var2;
  }
}
]]></code>
    </test-code>
```

**Comment by oowekyala** [comment]
In your test case the first field is already final so it's a true negative. We also don't check static fields (your second field) by design, although that could probably be relaxed. It's anyway an issue unrelated to this.
